bindings/python/pymongoarrow/context.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2021-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pyarrow import Table
+from pymongoarrow.lib import Int32Builder, Int64Builder, DoubleBuilder, DatetimeBuilder
+from pymongoarrow.types import _get_internal_typemap, _BsonArrowTypes
+
+
+_TYPE_TO_BUILDER_CLS = {
+    _BsonArrowTypes.int32: Int32Builder,
+    _BsonArrowTypes.int64: Int64Builder,
+    _BsonArrowTypes.double: DoubleBuilder,
+    _BsonArrowTypes.datetime: DatetimeBuilder
+}
+
+
+class PyMongoArrowContext:
```

Comment:
It would be helpful to add docstrings to describe the inputs/outputs/classes in this PR.

Reply:
It would be helpful to do the same for the Schema class too.

Reply:
Done.